packages/server/lib/gui/events.coffee

```diff
@@ -54,6 +54,9 @@ handleEvent = (options, bus, event, id, type, arg) ->
     when "on:config:changed"
       onBus("config:changed")
 
+    when "reload:configuration:requested"
+      onBus("reload:configuration:requested")
```

Comment:
I see where you're going with this - you're going to emit this event to the server, which will then immediately send it back to you (called the same thing).

It's a bit indirect but perhaps it's necessarily for reasons I don't see. I was thinking that you don't actually emit this event from the desktop-gui - instead you just call into the handler that you have set here: https://github.com/cypress-io/cypress/pull/3211/files/cd6a8e7f12e3e0cdeea5a8de4a1d40f44593f01b..b9e98f8f5a3f8eea5296561d96c3ec302c5c59a7#diff-63b2e970dc3b8d943b276fa3766e32adR183

to handle that, you just need to extract out the function into another function that's called when you click the "Reload" button in the desktop GUI or when the `ipc` event comes back to you